### PR TITLE
JAMES-3058 ADR: Cassandra mailbox object inconsistencies offline solving

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: src/adr/0020-cassandra-mailbox-object-consistency.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Failures during the denormalization process will lead to inconsistencies between the two tables.
+
+This can lead to the following user experience:
+
+```
+BOB creates mailbox A
+Denormalization fails and an error is returned to A
+
+BOB retry mailbox A creation
+BOB is being told mailbox A alread exists
```
Comment (Arsnael) [inline src/adr/0020-cassandra-mailbox-object-consistency.md]: s/alread/already

File: src/adr/0020-cassandra-mailbox-object-consistency.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+Failures during the denormalization process will lead to inconsistencies between the two tables.
+
+This can lead to the following user experience:
+
+```
+BOB creates mailbox A
+Denormalization fails and an error is returned to A
+
+BOB retry mailbox A creation
```
Comment (Arsnael) [inline src/adr/0020-cassandra-mailbox-object-consistency.md]: s/retry/retries

File: src/adr/0020-cassandra-mailbox-object-consistency.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+BOB is being told mailbox A alread exists
+
+BOB try to access mailbox A
```
Comment (Arsnael) [inline src/adr/0020-cassandra-mailbox-object-consistency.md]: s/try/tries

File: src/adr/0020-cassandra-mailbox-object-consistency.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+BOB is being told mailbox A do not exists
```
Comment (Arsnael) [inline src/adr/0020-cassandra-mailbox-object-consistency.md]: s/do no exists/does not exist

File: src/adr/0020-cassandra-mailbox-object-consistency.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+```
+BOB creates mailbox A
+Denormalization fails and an error is returned to A
+
+BOB retries mailbox A creation
+BOB is being told mailbox A already exists
+
+BOB tries to access mailbox A
+BOB is being told mailbox A does not exists
```
Comment (rouazana) [inline src/adr/0020-cassandra-mailbox-object-consistency.md]: s/exists/exist

File: src/adr/0020-cassandra-mailbox-object-consistency.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+## Decision
+
+We should provide an offline webadmin task to solve mailbox object inconsistencies.
```
Comment (rouazana) [inline src/adr/0020-cassandra-mailbox-object-consistency.md]: please define offline. as far as I know you cannot run any task with a stopped James

File: src/adr/0020-cassandra-mailbox-object-consistency.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+This task will read `mailbox` table and adapt path registrations in `mailboxPathV2`:
+ - Missing registrations will be added
+ - Orphan registrations will be removed
+ - Mismatch in content between the two table will require merging the two mailboxes together.
```
Comment (rouazana) [inline src/adr/0020-cassandra-mailbox-object-consistency.md]: table**s**

File: src/adr/0020-cassandra-mailbox-object-consistency.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+## Consequences
+
+As an administrator, if some of my users reports the bugs mentioned above, I have a way to sanitize my Cassandra 
+mailbox database.
+
+However, due to the two invariants mentioned above, we can not identify a clear source of trust based on existing 
+tables for the mailbox object. The task previously mentioned is subject to concurrency issues that might cancel 
+legitimate concurrent user actions.
+
+Hence this task must be run offline.
```
Comment (rouazana) [inline src/adr/0020-cassandra-mailbox-object-consistency.md]: idem

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**